sentry/src/main/java/io/sentry/config/ClasspathPropertiesLoader.java

```diff
@@ -0,0 +1,46 @@
+package io.sentry.config;
+
+import io.sentry.ILogger;
+import io.sentry.SentryLevel;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/** Loads {@link Properties} from the `sentry.properties` file located on the classpath. */
+final class ClasspathPropertiesLoader implements PropertiesLoader {
+  private final @NotNull String fileName;
+  private final @NotNull ClassLoader classLoader;
+  private final @NotNull ILogger logger;
+
+  public ClasspathPropertiesLoader(
+      @NotNull String fileName, @NotNull ClassLoader classLoader, @NotNull ILogger logger) {
+    this.fileName = fileName;
+    this.classLoader = classLoader;
+    this.logger = logger;
+  }
+
+  public ClasspathPropertiesLoader(@NotNull ILogger logger) {
+    this("sentry.properties", ClasspathPropertiesLoader.class.getClassLoader(), logger);
+  }
+
+  @Override
+  public @Nullable Properties load() {
+    try (InputStream inputStream = classLoader.getResourceAsStream(fileName)) {
```

Comment:
is it worth to wrap up on a buffered type?

Reply:
I may be missing something - I am not sure what do you mean. I used "try with resources" to ensure that the input stream gets closed.

Reply:
what I mean is to use a `BufferedInputStream`

Reply:
See https://github.com/getsentry/sentry-java/pull/946#discussion_r496467630

Reply:
I don't know so I will just follow your advice. This is read only once so I don't think there will be any difference in the performance. Is there any other benefit of using `BufferedInputStream`?

Reply:
it buffers to optimize the speed to access the data.
just a random blogpost https://pzemtsov.github.io/2015/01/19/on-the-benefits-of-stream-buffering-in-Java.html
its useful to make the App. startup faster (while initing the SDK), the downside is it could allocate a little bit more of memory.